designs/admin-api-health-status.md

```diff
@@ -0,0 +1,36 @@
+---
+state: draft | in-progress | complete
+---
+
+# Add Health and Status Endpoints to Admin API
+
+## Summary
+
+In addition to the available methods in the Admin API laid out [here](https://nodered.org/docs/api/admin/methods/), this document proposes the creation of `health` and `status` endpoints. These endpoints would be useful for application monitoring but could find broader use as well.
+
+While users can add this functionality manully to their own instances of Node-RED, this feature set feels sufficiently primitive and general that it would be useful to expose it as part of the Admin API.
+
+## Authors
+
+- [@szep](https://discourse.nodered.org/u/szep)
+
+## Details
+
+Two endpoints would be added to the Admin API:
+
+1. `GET /health` -- on success returns status code 200 and payload `{"status": "ok"}`
+   - Likely don't need to discuss what a fail condition would be since this endpoint is designed to indicate simply that Node-RED is up and running
+1. `GET /status` -- on success returns status code 200 and payload:
```

Comment:
Which status object ? They are currently for nodes.  Is this scoped to nodes ? in which case then does every node need to be able to provide status ? (imho no) - or to the overall runtime in which case what should it actually contain ?

Reply:
@dceejay, I'm imagining it returning an array of the same status objects that the status node provides. Only nodes that provide a status would be included in the return payload.

Reply:
none of the status is held as state anywhere apart from in the client browser - hence doing what you suggest now means we have to hold all that state centrally for all nodes just in case - whether there is a browser client or not. I think that is not something we want to do on the offchance someone wants it. I can see the use of an overall status of the runtime - eg maybe showing things like uptime, name of running flow, maybe process cpu load ? that sort of thing - but not state of every single node.

Reply:
> none of the status is held as state anywhere apart from in the client browser

That's not entirely true. The runtime does have to hold the current status value of all nodes so that when a new editor connects, it is able to deliver the status value to get it in sync.

That said, the node status is meant to be ephemeral internal state of the flow - used by the flow itself in its operation - it wasn't something intended to be exposed via admin apis. A flow could be created to expose node status if the flow author wants to do so - but that isn't a pattern I have ever seen mentioned in the community.

So it is technically possible to do, I would like to understand the specific use case for it a bit more.